q/verbose/verbose.q

```diff
@@ -0,0 +1,471 @@
+//type-checked equivalent to ! for creating a dictionary
+.finos.verbose.map:{[keylist;valuelist]
+    if[not type[keylist] within 0 98h; '"keylist must be a list"];
+    if[not type[valuelist] within 0 98h; '"valuelist must be a list"];
+    keylist!valuelist};
+
+//type-checked equivalent to ! for setting number of key columns in a table
+.finos.verbose.setKeyColNr:{[keycount;table]
+    if[not type[keycount] in -6 -7h; '"keycount must be an integer"];
+    if[0>keycount; '"keycount must be nonnegative"];
+    if[not type[table] in 98 99h; '"2nd argument must be a table"];
+    if[99h=type[table];
+        if[any not 98h=type[key table],type[value table]; '"2nd argument must be a table"];
+    ];
+    keycount!table};
+
+//workaround for inability to use table syntax with security.q
+.finos.verbose.table:{[keyCols;valueCols]
+    if[not 0h=type keyCols; '"column name-values must be specified as a list"];
+    if[not 0h=type valueCols; '"column name-values must be specified as a list"];
+    if[0<count keyCols; :.z.s[();keyCols]!.z.s[();valueCols]];
+    if[not 0=count[valueCols]mod 2; '"column name-value list must have even number of elements"];
+    colc:count[valueCols] div 2;
+    if[not 11h=type coln:valueCols[2*til colc]; '"column names must be symbols"];
+    colv:valueCols[1+2*til colc];
+    if[any not (type each colv) within 0 97h; '"column values must be lists"];
+    flip coln!colv};
+
+.finos.verbose.sym:{[str]
+    if[not 10h=type str; '".finos.verbose.sym only works on string argument"];
+    `$str};
+
+//wrapper to allow xasc in secure gateway (raw xasc may modify table in-place)
```

Comment:
"secure gateway" is a stranded reference. Repeated multiple times.

Reply:
Also changed to refer to "parse tree filters".